src/gasPriceOracle/oracle.ts

```diff
@@ -0,0 +1,98 @@
+import { BigNumber, providers } from "ethers";
+import { toBN } from "../utils";
+
+export type GasPriceEstimate = {
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+};
+
+type ValidatedFeeData = {
+  [idx: string]: BigNumber;
+  gasPrice: BigNumber;
+  maxFeePerGas: BigNumber;
+  maxPriorityFeePerGas: BigNumber;
+};
+
+interface GasPriceFeed {
+  (provider: providers.Provider): Promise<GasPriceEstimate>;
+}
+
+async function getProviderFeeData(provider: providers.Provider): Promise<ValidatedFeeData> {
+  // Start with safe defaults.
+  const feeData: ValidatedFeeData = {
+    gasPrice: toBN(Number.MAX_SAFE_INTEGER),
+    maxFeePerGas: toBN(Number.MAX_SAFE_INTEGER),
+    maxPriorityFeePerGas: toBN(Number.MAX_SAFE_INTEGER),
```

Comment:
Why do we default these to a large integer instead of making them optional?

Reply:
To me, it seems safer to default to a large number in order to ensure that a gas price resolution failure doesn't accidentally make a transaction look profitable. This has the advantage of being drop-in compatible with the calling code, but has the disadvantage of suppressing price lookup failures (could be mitigated if we add logging here). As I see it, the alternative would be to throw exceptions from this code in the event of gas price lookup failures, which is a more explicit communication to the caller that something went wrong.

Extrapolating the impact of a "default high" price into the two consumers of this functionality:
- In the bots, this should be fine because they only use this to filter out unprofitable transactions.
- On the front end, this will prevent users from placing transactions, and does indirectly mask the underlying error, which _might_ make Across look uncompetitive on the rare occasion that this does happen.

In terms of the selection of MAX_SAFE_INTEGER, it is just an easy constant to use here, but it doesn't have to be this specific constant. It does occur to me that it's wasteful to take the number MAX_SAFE_INTEGER, convert it to a string, and then convert it to a BigNumber.

If we choose to proceed with a high default, we'd probably also be fine to use 1 full unit of the underlying gas token (i.e. 1 ETH, 1 MATIC). That might become more complicated if we ever have a gas token that's not 18 decimals, but I guess that's unlikely.

Reply:
nb The change implemented in this PR is a departure from the current norm, so I need to confirm what impact it might have in the FE (i.e. if the FE specifically looks for a 0 response then that might be an issue...).